agent/consul/server.go

```diff
@@ -595,14 +595,13 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server,
 	})
 
 	s.hcpManager = hcp.NewManager(hcp.ManagerConfig{
-		CloudConfig:       s.config.Cloud,
-		Client:            flat.HCP.Client,
+		CloudConfig:       flat.HCP.Config,
 		StatusFn:          s.hcpServerStatus(flat),
 		Logger:            logger.Named("hcp_manager"),
 		SCADAProvider:     flat.HCP.Provider,
 		TelemetryProvider: flat.HCP.TelemetryProvider,
 		ManagementTokenUpserterFn: func(name, secretId string) error {
-			if s.IsLeader() {
+			if s.config.ACLsEnabled && s.IsLeader() {
```

Comment:
Not sure I've seen the reasoning behind this change, can we clarify?

Reply:
This was missed by me when I originally added this — if ACLs are not enabled, `upsertManagementToken` will error, so we'd end up with unwanted error logs.

Reply:
@mkam happy to look into this myself in a followup, but just wanted to bring it up: maybe we also want to check `s.InPrimaryDatacenter() `  so we are being consistent with how this works: https://github.com/hashicorp/consul/blob/37a5fddffa37e33036089074c0a1b5b071096e1a/agent/consul/leader.go#L440